sky/backends/backend_utils.py

```diff
@@ -1854,13 +1877,15 @@ def run_ray_status_to_check_ray_cluster_healthy() -> bool:
                 f'{colorama.Style.RESET_ALL}')
     assert len(node_statuses) <= handle.launched_nodes
 
-    # If the node_statuses is empty, all the nodes are terminated. We can
-    # safely set the cluster status to TERMINATED. This handles the edge case
-    # where the cluster is terminated by the user manually through the UI.
-    to_terminate = not node_statuses
+    # If the node_statuses is empty, it should mean that all the nodes are
+    # terminated and we can set the cluster status to TERMINATED. This handles
+    # the edge case where the cluster is terminated by the user manually through
+    # the UI.
+    to_terminate = len(node_statuses) == 0
```

Comment:
style: https://google.github.io/styleguide/pyguide.html#2144-decision
```suggestion
    to_terminate = not node_statuses
```